cpp/src/Slice/Parser.cpp

```diff
@@ -4178,6 +4178,13 @@ Slice::Operation::createParameter(const string& name, const TypePtr& type, bool
 
     if (isOutParam)
     {
+        // If the operation returns multiple values, but none of those are out parameters, it must be a return-tuple,
+        // in which case using out parameters isn't allowed, and an error is thrown.
+        if (!_usesOutParameters && _returnValues.size() > 1)
+        {
+            _unit->error("operations using return-tuples cannot use out parameters");
```

Comment:
I think `an operation returning a tuple cannot have out parameters` is more readable.

Reply:
Sounds good to me!